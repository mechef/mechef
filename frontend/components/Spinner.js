// @flow

import * as React from 'react';

const Spinner = () => (
  <div className="spinnerOverlay">
    <div className="spinner" />
    <style jsx>
      {`
        .spinnerOverlay {
          position: fixed;
          top: 0;
          left: 0;
          z-index: 1;
          width: 100vw;
          height: 100vh;
          background-color: rgba(0, 0, 0, 0.6);
        }
        .spinner {
          position: fixed;
          z-index: 99;
          top: 50%;
          left: 50%;
          width: 40px;
          height: 40px;
          background-color: #ffffff;
          background-image: url('../static/svg/loading_02.svg');
          animation: sk-rotateplane 1.2s infinite ease-in-out;
        }

        @keyframes sk-rotateplane {
          0% {
            transform: perspective(120px) rotateX(0deg) rotateY(0deg);
            -webkit-transform: perspective(120px) rotateX(0deg) rotateY(0deg)
          } 50% {
            transform: perspective(120px) rotateX(-180.1deg) rotateY(0deg);
            -webkit-transform: perspective(120px) rotateX(-180.1deg) rotateY(0deg)
          } 100% {
            transform: perspective(120px) rotateX(-180deg) rotateY(-179.9deg);
            -webkit-transform: perspective(120px) rotateX(-180deg) rotateY(-179.9deg);
          }
        }
      `}
    </style>
  </div>
);

export default Spinner;
