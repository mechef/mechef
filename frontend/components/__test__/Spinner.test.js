import renderer from 'react-test-renderer';
import React from 'react';

import Spinner from '../Spinner';


describe('Pages With Snapshot Testing', () => {
  it('snapshot Spinner Component', () => {
    const component = renderer.create(<Spinner />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
