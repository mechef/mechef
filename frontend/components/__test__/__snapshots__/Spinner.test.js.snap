// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages With Snapshot Testing snapshot Spinner Component 1`] = `
<div
  className="jsx-3364859594 spinnerOverlay"
>
  <div
    className="jsx-3364859594 spinner"
  />
</div>
`;
